rust/chains/hyperlane-cosmos/src/mailbox.rs

```diff
@@ -0,0 +1,378 @@
+    }
+
+    #[instrument(err, ret, skip(self))]
+    async fn default_ism(&self) -> ChainResult<H256> {
+        let payload = mailbox::DefaultIsmRequest {
+            default_ism: general::EmptyStruct {},
+        };
+
+        let data = self
+            .provider
+            .wasm_query(GeneralMailboxQuery { mailbox: payload }, None)
+            .await?;
+        let response: mailbox::DefaultIsmResponse = serde_json::from_slice(&data)?;
+
+        // convert Hex to H256
+        let ism = H256::from_slice(&hex::decode(response.default_ism)?);
+        Ok(ism)
+    }
+
+    #[instrument(err, ret, skip(self))]
+    async fn recipient_ism(&self, recipient: H256) -> ChainResult<H256> {
+        let address = verify::digest_to_addr(recipient, &self.signer.prefix)?;
+
+        let payload = mailbox::RecipientIsmRequest {
+            recipient_ism: mailbox::RecipientIsmRequestInner {
+                recipient_addr: address,
+            },
+        };
+
+        let data = self
+            .provider
+            .wasm_query(GeneralMailboxQuery { mailbox: payload }, None)
+            .await?;
+        let response: mailbox::RecipientIsmResponse = serde_json::from_slice(&data)?;
+
+        // convert Hex to H256
+        let ism = verify::bech32_decode(response.ism);
+        Ok(ism)
+    }
+
+    #[instrument(err, ret, skip(self))]
+    async fn process(
+        &self,
+        message: &HyperlaneMessage,
+        metadata: &[u8],
+        tx_gas_limit: Option<U256>,
+    ) -> ChainResult<TxOutcome> {
+        let process_message = ProcessMessageRequest {
+            process: ProcessMessageRequestInner {
+                message: hex::encode(RawHyperlaneMessage::from(message)),
+                metadata: hex::encode(metadata),
+            },
+        };
+
+        let response: TxResponse = self
+            .provider
+            .wasm_send(process_message, tx_gas_limit)
+            .await?;
+        Ok(TxOutcome {
+            transaction_id: h256_to_h512(H256::from_slice(
+                hex::decode(response.txhash).unwrap().as_slice(),
+            )),
+            executed: response.code == 0,
+            gas_used: U256::from(response.gas_used),
+            gas_price: U256::from(response.gas_wanted),
+        })
+    }
+
+    #[instrument(err, ret, skip(self), fields(msg=%message, metadata=%fmt_bytes(metadata)))]
+    async fn process_estimate_costs(
+        &self,
+        message: &HyperlaneMessage,
+        metadata: &[u8],
+    ) -> ChainResult<TxCostEstimate> {
+        let process_message = ProcessMessageRequest {
+            process: ProcessMessageRequestInner {
+                message: hex::encode(RawHyperlaneMessage::from(message)),
+                metadata: hex::encode(metadata),
+            },
+        };
+
+        let response: SimulateResponse = self.provider.wasm_simulate(process_message).await?;
+        let result = TxCostEstimate {
+            gas_limit: U256::from(response.gas_info.unwrap().gas_used),
```

Comment:
shouldn't unwrap here, thinking we should have a commit that removes all unwraps in the cosmos crates to avoid missing any of them